src/plugins/data/public/search/session/lib/session_name_formatter.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import moment from 'moment';
+
+export function formatSessionName(
+  sessionName: string,
+  opts: { sessionStartTime?: Date; appendStartTime?: boolean }
+): string {
+  if (opts.sessionStartTime && opts.appendStartTime) {
```

Comment:
Why do we want to make this configurable using `opts.appendStartTime`?
Why not add the start time if it's available?

Reply:
I am not 100% that automatically appending a name is a good thing in the first place. 
that's mostly the reason why I decided to at least allow to opt out

Reply:
I'd rather have @elastic-jb decide one way or the other, 
otherwise we'd need to make a UI for this and maintain and test this option.
Feels a bit unnecessary to me.

Reply:
> I am not 100% that automatically appending a name is a good thing in the first place.
that's mostly the reason why I decided to at least allow to opt out

What is your concern/argument against?

Reply:
I think I understand - 
If lets say ML has a custom name with a unique identifier, they might not want the date added.

Reply:
> If lets say ML has a custom name with a unique identifier, they might not want the date added.

yes, thanks!

also: 
I am a tiny bit concerned that with the default name we should date twice in mgmt table. See how time is part of a name and part of a column: 

![Screen Shot 2021-03-18 at 13 09 28](https://user-images.githubusercontent.com/7784120/111623894-46cf7f00-87eb-11eb-895e-7ffd888974be.png)

If not for this tiny quirk, I'd have zero concerns :D

Reply:
Oh well :) :+1:

Reply:
👍 makes sense.  So our 3 options for when someone saves a session are:
1) Append the date (which seems to have multiple downsides — length of title, duplicate date, and unwanted in some cases)
2) Append a number `Discover (1)` or `Discover - 2` or `Discover 3` (I'd prefer the parenthesis as it's more recognizable that this was added on
3) Do nothing (Do we run into name conflicts? What are the downsides of this approach?)

Reply:
@mdefazio The number approach would be great, but it would require detecting duplicates, which is not trivial.

Reply:
@mdefazio, I love the date approach and I think we can live with that minor quirk